### PR TITLE
Fix-up and document `EventEmitter` example.

### PR DESCRIPTION
**Pull request by kjvalencik**
Resolves https://github.com/neon-bindings/examples/issues/27

- Remove unrelated code
- Implement receiver timeout
- Document
  * Design
  * Limitations
  * Code

Since this removes the express example, I'll follow-up with a separate PR to add that back in a separate example. I'm not sure if the express example should include `neon-serde` or if that should be a distinct example.

### CODE REVIEW

**Comment by amilajack** [comment]
Failing because of [linter errors](https://travis-ci.org/neon-bindings/examples/jobs/492790028#L907). Most of these can be fixed with `yarn lint --fix`.

File: event-emitter/lib/index.js [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
-    .catch(() => console.log('port in use'));
+  // Shutdown the emitter after 5s.
+  return new Promise(resolve => setTimeout(resolve, 5000))
+    .then(() => emitter.shutdown());
 }
 
-module.exports = require('../native/index.node');
+if (process.env.NODE_ENV !== 'test') {
+  run();
+}
```
Comment (amilajack) [inline event-emitter/lib/index.js:17]:
```suggestion
}

module.exports = run;
```

Comment (amilajack) [reply event-emitter/lib/index.js:17]: To keep this example consistent with all the other examples, we need to export run from `./lib/index.js`

**Review by amilajack — CHANGES_REQUESTED** [review]

**Comment by kjvalencik** [comment]
@amilajack FYI, one of eslint options is incorrect.

```
'use strict' is unnecessary inside of modules
```

I think eslint needs to be configured to not expect modules since these aren't.

**Comment by kjvalencik** [comment]
Updated and passing tests.

**Comment by amilajack** [comment]
Awesome!

**Pull request merged**